File: src/TwoPointers/lc0011_ContainerWithMostWater.py
# https://leetcode.cn/problems/container-with-most-water
def container_with_most_water(height: list[int]) -> int:
    if len(height) <= 1:
        raise "Invalid Input height"
    left_index = 0
    right_index = len(height) - 1
    max_area = 0
    while True:
        if left_index == right_index:
            break
        area = (right_index - left_index) * min(height[left_index], height[right_index])
        max_area = area if area > max_area else max_area
        if height[left_index] < height[right_index]:
            left_index += 1
        else:
            right_index -= 1
    return max_area

File: src/TwoPointers/lc0011_ContainerWithMostWater.go
package TwoPointers

// https://leetcode.cn/problems/container-with-most-water
func containerWithMostWater(height []int) int {

	if len(height) <= 1 {
		panic("Invalid Input height")
	}
	leftIndex := 0
	rightIndex := len(height) - 1
	maxArea := 0
	for {
		if leftIndex == rightIndex {
			break
		}
		area := (rightIndex - leftIndex) * min(height[leftIndex], height[rightIndex])
		if area > maxArea {
			maxArea = area
		}
		if height[leftIndex] < height[rightIndex] {
			leftIndex += 1
		} else {
			rightIndex -= 1
		}
	}
	return maxArea
}
